fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/cache/LogicalResourceIdentCacheImpl.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * (C) Copyright IBM Corp. 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence.jdbc.cache;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.ibm.fhir.persistence.jdbc.dao.api.ILogicalResourceIdentCache;
+import com.ibm.fhir.persistence.jdbc.dao.api.LogicalResourceIdentKey;
+import com.ibm.fhir.persistence.jdbc.dao.impl.ResourceReferenceValueRec;
+
+
+/**
+ * Implementation of a cache used for lookups of entities related
+ * to local and external resource references
+ */
+public class LogicalResourceIdentCacheImpl implements ILogicalResourceIdentCache {
+
+    // We use LinkedHashMap for the local map because we also need to maintain order
+    // of insertion to make sure we have correct LRU behavior when updating the shared cache
+    private final ThreadLocal<LinkedHashMap<LogicalResourceIdentKey, Long>> localLogicalResourceIdents = new ThreadLocal<>();
+
+    // The lru token values cache shared at the server level
+    private final LRUCache<LogicalResourceIdentKey, Long> cache;
```

Comment:
when you say "shared at the server level" ...its still isolated per tenant / per datastore, right?

Reply:
ok, i think I confirmed this in the code... "server level" just means its shared across all threads / all requests (unlike the threadlocal above).